tests/integration_tests/security/test_jail.py

```diff
@@ -581,3 +581,34 @@ def test_new_pid_namespace(uvm_plain):
     assert len(nstgid_list) == 2
     assert int(nstgid_list[1]) == 1
     assert int(nstgid_list[0]) == fc_pid
+
+
+@pytest.mark.parametrize(
+    "daemonize",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "new_pid_ns",
+    [True, False],
+)
+def test_firecracker_kill_by_pid(
+    microvm_factory, guest_kernel_linux_5_10, rootfs_ubuntu_22, daemonize, new_pid_ns
+):
+    """
+    Test that Firecracker is spawned in a new PID namespace if requested.
+    """
+    microvm = microvm_factory.build(
+        guest_kernel_linux_5_10,
+        rootfs_ubuntu_22,
+    )
+    microvm.jailer.daemonize = daemonize
+    microvm.jailer.new_pid_ns = new_pid_ns
```

Comment:
Just set this after `spawn` so it does not get overridden

Reply:
Ah right. I feel there should be no need to add new options. There is `jailer_kwargs` to specify exactly what options the jailer should use.

Reply:
We cannot pass new_pid_ns via jailer_kwargs because we call JailerContext with new_pid_ns=True [here](https://github.com/firecracker-microvm/firecracker/blob/fc674a38817c044cde6b2ec8a90fbe15caf27a4f/tests/framework/microvm.py#L186) 
so it will return `multiple values for keyword argument 'new_pid_ns'` error.

For daemonize I chose this way so that its consistent with the other tests code we have in other places plus I think now that I changed code to use uvm_plain we don't have the option to use daemonize via the jailer_kwargs

Reply:
> We cannot pass new_pid_ns via jailer_kwargs because we call JailerContext with new_pid_ns=True [here](https://github.com/firecracker-microvm/firecracker/blob/fc674a38817c044cde6b2ec8a90fbe15caf27a4f/tests/framework/microvm.py#L186)
so it will return multiple values for keyword argument 'new_pid_ns' error.

That can easily be solved like so:

```
    ...
    new_pid_ns=jailer_kwargs.pop("new_pid_ns", True)
```

alternatively, change the JailerContext default to `True` and then remove the True in Microvm.